### ISSUE TITLE
404 errors when getting module info from Terraform Cloud registry

### ISSUE THREAD

**Issue by aclarknexient** [hashicorp/terraform-ls#1229]
# 404 errors when getting module info from Terraform Cloud registry

While using VSCode extension preview, I get the following errors when trying to access our registry:

```
2023/03/25 10:16:36 jobs.go:383: JOBS: Finishing job "211": "OpTypeGetModuleDataFromRegistry" for {"file:///redacted/redacted/redacted/tf/modules/infra"} (err = 4 errors occurred:
	* unexpected response 404 Not Found: {"errors":["Not Found"]}

	* unexpected response 404 Not Found: {"errors":["Not Found"]}

	* unexpected response 404 Not Found: {"errors":["Not Found"]}

	* unexpected response 404 Not Found: {"errors":["Not Found"]}


, deferredJobs: [])
```

## Terraform Version
```
1.3.6
```

## Terraform LS version
```
0.30.3
```

## Client Version
```
/REDACTED/.vscode/extensions/hashicorp.terraform-preview-2023.2.2216-darwin-x64/bin/terraform-ls
```

## Expected Behavior
The module information should have been fetched.

## Actual Behavior
404 not found error

## Question
Is there a way to tell which urls are being attempted by the language server? I want to try to replicate this on the command line with curl to see if there's a token that might be needed, or whether a proxy is being used.

## Other information
When I commented out the module source attributes, the 404 errors went away. The pause of 30 seconds or so when saving+formatting has also disappeared.

**Comment by aclarknexient** [comment]
This happens around the same time that the document finally formats and saves about 10 seconds after I hit cmd+s to save.

**Comment by aclarknexient** [comment]
@xiehan this appears to be fixed with the latest version 0.31.1 -- should I close this ticket now? I will update it if I run into any more problems.

**Comment by dbanck** [comment]
@aclarknexient That's great to hear! You are right; we just released 0.31.1 with a fix for this earlier.

I will close this for now, but if this is still an issue, please respond, and we'll re-open it.
